File: codegen/results/v001/qwen3.5_27b_nothink/20260408_102056/typescript/003_fibonacci/solution.ts
const main = () => {
  const args = process.argv.slice(2);
  
  if (args.length === 0) {
    return;
  }

  const n = parseInt(args[0], 10);

  if (isNaN(n) || n < 1) {
    return;
  }

  let a = 1;
  let b = 1;

  // Print the first number if it's within range
  if (a <= n) {
    console.log(a);
  } else {
    return;
  }

  // Continue generating Fibonacci numbers
  while (b <= n) {
    console.log(b);
    const next = a + b;
    a = b;
    b = next;
  }
};

main();

File: codegen/results/v001/qwen3.5_27b_nothink/20260408_102056/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	// Ensure exactly one command-line argument is provided
	if len(os.Args) != 2 {
		return
	}

	// Parse the integer argument N
	n, err := strconv.ParseInt(os.Args[1], 10, 64)
	if err != nil {
		return
	}

	// If N < 1, print nothing
	if n < 1 {
		return
	}

	// Initialize the first two Fibonacci numbers
	a := int64(1)
	b := int64(1)

	// Print the first number if it's <= N
	if a <= n {
		fmt.Println(a)
	}

	// If N is 1, we stop here because the next number is also 1,
	// but the sequence logic below handles the second 1 correctly
	// by checking b <= n. However, to strictly follow the sequence
	// 1, 1, 2... we need to ensure the second 1 is printed if N >= 1.
	if b <= n {
		fmt.Println(b)
	}

	// Generate subsequent Fibonacci numbers
	for {
		next := a + b
		if next > n {
			break
		}
		fmt.Println(next)
		a = b
		b = next
	}
}